ui-package/src/components/SubmitConsolidationsForm/ConsolidationReview.tsx

```diff
@@ -0,0 +1,217 @@
+            </div>
+            <div className="col-9 col-lg-10">
+              <input type="checkbox" className="form-check-input" id="addExtraFee" checked={addExtraFee} onChange={(e) => setAddExtraFee(e.target.checked)} />
+              <label htmlFor="addExtraFee" className="ms-1">Add extra fee to avoid rejection due to other submissions</label>
+            </div>
+          </div>
+          <div className="row">
+            <div className="col-3 col-lg-2">
+              Total fee:
+            </div>
+            <div className="col-9 col-lg-10">
+              {toReadableAmount(requestFee, feeFactor, feeUnit, 4)}
+            </div>
+          </div>
+          <div className="row mt-3">
+            <div className="col-12">
+              <button className="btn btn-primary" disabled={submitRequest.isPending || submitRequest.isSuccess} onClick={() => submitConsolidation()}>
+                {submitRequest.isSuccess ?
+                  <span>Submitted</span> :
+                  submitRequest.isPending ? (
+                    <span className="text-nowrap"><div className="spinner-border spinner-border-sm me-1" role="status"></div>Pending...</span>
+                    ) : (
+                      submitRequest.isError ? (
+                        <span className="text-nowrap"><i className="fa-solid fa-repeat me-1"></i> Retry consolidation</span>
+                      ) : (
+                        "Submit consolidation request"
+                      )
+                    )
+                }
+              </button>
+            </div>
+          </div>
+          {submitRequest.isSuccess ?
+            <div className="row mt-3">
+              <div className="col-12">
+                <div className="alert alert-success">
+                  Consolidation TX: 
+                  {props.explorerUrl ?
+                    <a className="ms-1" href={props.explorerUrl + "tx/" + submitRequest.data} target="_blank" rel="noreferrer">{submitRequest.data}</a>
+                  : <span className="ms-1">{submitRequest.data}</span>
+                  }
+                </div>
+              </div>
+            </div>
+          : null}
+        </div>
+        }
+        {errorModal && (
+          <Modal show={true} onHide={() => setErrorModal(null)} size="lg">
+            <Modal.Header closeButton>
+              <Modal.Title>Consolidation Transaction Failed</Modal.Title>
+            </Modal.Header>
+            <Modal.Body>
+              <pre className="m-0">{errorModal}</pre>
+            </Modal.Body>
+            <Modal.Footer>
+              <button className="btn btn-primary" onClick={() => setErrorModal(null)}>Close</button>
+            </Modal.Footer>
+          </Modal>
+        )}
+    </div>
+  );
+
+  function getRequiredFee(numerator: bigint): bigint {
+    // https://eips.ethereum.org/EIPS/eip-7251#fee-calculation
+    let i = 1n;
+    let output = 0n;
+    let numeratorAccum = 1n * 17n; // factor * denominator
```

Comment:
Maybe worth to just create a const/var for this 17? It's from `CONSOLIDATION_REQUEST_FEE_UPDATE_FRACTION` right? Ideally we would probably read it from config, but it can get annoying passing that to the UI code.